graph/converter.py

```diff
@@ -0,0 +1,357 @@
+import graph.oval_graph
+import collections
+import re
+import uuid
+
+
+class converter():
+    def __init__(self, tree):
+        if isinstance(tree, graph.oval_graph.OvalNode):
+            self.tree = tree
+        else:
+            raise ValueError('err - this is not tree created from OvalNodes')
+
+    def _get_node_icon(self):
+        value = self.tree.evaluate_tree()
+        icon = None
+        if value is None:
+            if self._is_negated_boolean('true', self.tree.value):
+                icon = 'false'
+            elif self._is_negated_boolean('false', self.tree.value):
+                icon = 'true'
+            else:
+                icon = self.tree.value
+            icon, value = self.tree.value, icon
+        else:
+            if self._is_negated_boolean('true', value):
+                icon = 'false'
+            elif self._is_negated_boolean('false', value):
+                icon = 'true'
+            else:
+                icon = value
+
+        VALUE_TO_COLOR = {
```

Comment:
If dictionaries like this are renamed and moved to class top level, it will make style overrides even easier.